src/nvim/decoration.c

```diff
@@ -124,6 +126,13 @@ void decor_redraw_sh(buf_T *buf, int row1, int row2, DecorSignHighlight sh)
       redraw_buf_range_later(buf, row1 + 1, row2 + 1);
     }
   }
+  if (sh.flags & kSHConcealLines) {
+    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+      if (wp->w_buffer == buf) {
+        changed_window_setting(wp);
```

Comment:
TODO comment: this forces a redraw off the entire window. we want something more fine grained later.

Reply:
Yep, also see https://github.com/neovim/neovim/pull/31324#discussion_r1857475348.